File: queue.cpp
#include <iostream>
const int size=5;
using namespace std;
class Queue{
    int arr[size];
    int front,rear;
    public:    
    Queue(){
        front,rear=-1;
    }    
    void Enqueue(int data){ 
        if(rear==size-1){
            cout<<"Queue is Full! "<<endl;
        }
        else{
            rear++; 
            arr[rear]=data;
            front=arr[0];
            cout<<"Data Inserted! ";
        }
    }
    void Dequeue(){
        if(front==-1 || front>rear){
            cout<<"Queue is Empty! "<<endl;
        }
        else{
            // arr[front]=NULL;
            front++;
            cout<<"Element deleted! ";
        }
    }
    void Display(){
        if(front==-1){
            cout<<"Queue is Empty";
        }
        else{
            cout<<"The Queue : ";
            for(int i=front-1;i<=rear;i++){
                cout<<arr[i]<<" ";
            }
        }
    }
};
int main(){
    Queue q;
    int ch,data;
    do{
        cout<<endl<<"Menu"<<endl;
        cout<<"1. Enqueue "<<endl;
        cout<<"2. Dequeue "<<endl;
        cout<<"3. Display the Queue "<<endl;
        cout<<"4. Exit Menu"<<endl;
        cout<<"Enter your choice: ";
        cin>>ch;
        switch(ch){
            case 1: cout<<"Enter the value to be inserted: ";cin>>data; q.Enqueue(data);break;
            case 2: q.Dequeue();break;
            case 3: q.Display();break;
            case 4: cout<<"Menu exited!";break;
            default: cout<<"Invalid Case!"<<endl;
        }
    }
    while(ch!=4);
}

File: queue.java
import java.util.Scanner;
public class Queue{
    static int size=5;
    static int arr[]=new int[size];
    static int front=0;
    static int rear=0;
    public static void Enqueue(int data) {
        if(rear==size){
            System.out.println("Queue is Full ! ");
        }
        else{
            arr[rear]=data;
            rear++;
            System.out.println("Element Inserted! ");
        }
    }
    public static void Dequeue() {
        if(front==size){
            System.out.println("Queue is Empty! ");
        }
        else{
            front++;
            System.out.println("Element removed! ");
        }
    }
    public static void Display() {
        System.out.println("The Queue: ");
        for(int i=front;i<size;i++){
            System.out.println(arr[i]);
        }
    }
    public static void main(String[] args){
        int ch,data;
        do{
            System.out.println("Menu");
            System.out.println("1. Enqueue ");
            System.out.println("2. Dequeue ");
            System.out.println("3. Display ");
            System.out.println("4. Exit ");
            System.out.println("Enter your choice: ");
            Scanner sc = new Scanner(System.in);
            ch= sc.nextInt();
            switch (ch) {
                case 1:
                    System.out.println("Enter the Value to be Inserted: ");
                    data=sc.nextInt();
                    Enqueue(data);
                    break;
                case 2: Dequeue();break;
                case 3: Display();break;
                case 4: System.out.println("Menu Exited");break;
                default: System.out.println("Invalid Choice ");
                sc.close();
            }
        }
        while(ch!=4);
    }
}
